web_src/js/index.js

```diff
@@ -109,8 +109,22 @@ function initBranchSelector() {
   const $branchMenu = $selectBranch.find('.reference-list-menu');
   $branchMenu.find('.item:not(.no-select)').click(function () {
     const selectedValue = $(this).data('id');
+    const editMode = $('#editing_mode').val();
     $($(this).data('id-selector')).val(selectedValue);
-    $selectBranch.find('.ui .branch-name').text(selectedValue);
+
+    if (editMode === 'true') {
+      const form = $('#update_issueref_form');
+
+      $.post(form.attr('action'), {
+        _csrf: csrf,
+        ref: selectedValue
+      },
+      () => {
+        window.location.reload();
```

Comment:
I think we could not reload it and just change the value.

Reply:
This could be done in other PR along with assignee, label and milestone change